rubrics/tool/src/main/java/org/sakaiproject/rubrics/RubricsApplication.java

```diff
@@ -0,0 +1,96 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ **********************************************************************************/
+
+package org.sakaiproject.rubrics;
+
+import java.util.Arrays;
+import javax.servlet.DispatcherType;
+import javax.servlet.ServletContext;
+import javax.servlet.ServletException;
+
+import lombok.extern.slf4j.Slf4j;
+import org.sakaiproject.component.cover.ComponentManager;
+import org.sakaiproject.component.impl.SpringCompMgr;
+import org.sakaiproject.util.RequestFilter;
+import org.sakaiproject.util.ToolListener;
+import org.springframework.boot.Banner;
+import org.springframework.boot.autoconfigure.EnableAutoConfiguration;
+import org.springframework.boot.autoconfigure.SpringBootApplication;
+import org.springframework.boot.autoconfigure.jdbc.DataSourceAutoConfiguration;
+import org.springframework.boot.autoconfigure.orm.jpa.HibernateJpaAutoConfiguration;
+import org.springframework.boot.builder.SpringApplicationBuilder;
+import org.springframework.boot.web.servlet.FilterRegistrationBean;
+import org.springframework.boot.web.servlet.ServletRegistrationBean;
+import org.springframework.boot.web.support.SpringBootServletInitializer;
+import org.springframework.context.ConfigurableApplicationContext;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.web.context.support.AnnotationConfigWebApplicationContext;
+import org.springframework.web.servlet.DispatcherServlet;
+
+@Configuration
+@EnableAutoConfiguration(exclude={DataSourceAutoConfiguration.class, HibernateJpaAutoConfiguration.class})
+@SpringBootApplication
+@Slf4j
+public class RubricsApplication extends SpringBootServletInitializer {
+
+    /**
+     * Required per http://docs.spring.io/spring-boot/docs/current/reference/html/howto-traditional-deployment.html
+     * in order produce a traditional deployable war file.
+     *
+     * @param application
+     * @return
+     */
+
+    @Override
+    protected SpringApplicationBuilder configure(SpringApplicationBuilder application) {
+        ConfigurableApplicationContext sharedAc = ((SpringCompMgr) ComponentManager.getInstance()).getApplicationContext();
+        application.parent(sharedAc);
+        return application.bannerMode(Banner.Mode.OFF).sources(RubricsApplication.class);
+    }
+
+    @Bean
+    public ServletRegistrationBean servletRegistrationBean() {
+        ServletRegistrationBean srb = new ServletRegistrationBean(new DispatcherServlet(new AnnotationConfigWebApplicationContext())) {
+            @Override
+            public void onStartup(ServletContext servletContext) throws ServletException {
+                super.onStartup(servletContext);
+                servletContext.addListener(ToolListener.class);
+                // servletContext.addListener(SakaiContextLoaderListener.class);
```

Comment:
This might be the cause of the NPE we're getting, I guess we need to include it.